sdk/cosmos/azure-cosmos/azure/cosmos/_execution_context/document_producer.py

```diff
@@ -276,3 +276,57 @@ def _validate_orderby_items(self, res1, res2):
             type2 = _OrderByHelper.getTypeStr(elt2)
             if type1 != type2:
                 raise ValueError("Expected {}, but got {}.".format(type1, type2))
+
+
+class _NonStreamingDocumentProducer(object):
+    """This class takes care of handling of the items to be sorted in a non-streaming context.
+
+    One instance of this document producer goes attached to every item coming in for the priority queue to be able
+    to properly sort items as they get inserted.
+    """
+
+    def __init__(self, item_result, sort_order):
+        """
+        Constructor
+        """
+        self._item_result = item_result
+        self._doc_producer_comp = _NonStreamingOrderByComparator(sort_order)
+
+    def __lt__(self, other):
+        return self._doc_producer_comp.compare(self, other) < 0
+
+
+class _NonStreamingOrderByComparator(object):
+    """Provide a Comparator for item results which respects orderby sort order.
+    """
+
+    def __init__(self, sort_order):  # pylint: disable=super-init-not-called
+        """Instantiates this class
+
+        :param list sort_order:
+            List of sort orders (i.e., Ascending, Descending)
+
+        :ivar list sort_order:
+            List of sort orders (i.e., Ascending, Descending)
+
+        """
+        self._sort_order = sort_order
+
+    def compare(self, doc_producer1, doc_producer2):
```

Comment:
>compare(self, doc_producer1, doc_producer2): [](http://example.com/codeflow?start=8&length=44)

This seems incorrect... in the non streaming case, we compare _items_ not document producers. Also, I dont think you would need a new comparator in this case since you can just use `_OrderByHelper.compare`